tests/platform_tests/api/test_psu.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import re
+
+import pytest
+import yaml
+
+from common.helpers.assertions import pytest_assert
+from common.helpers.platform_api import psu
+from common.helpers.platform_api import chassis
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.sanity_check(skip_sanity=True),
+    pytest.mark.disable_loganalyzer  # disable automatic loganalyzer
+]
+
+STATUS_LED_COLOR_GREEN = "green"
+STATUS_LED_COLOR_AMBER = "amber"
+STATUS_LED_COLOR_RED = "red"
+STATUS_LED_COLOR_OFF = "off"
+
+class TestPSUAPI(object):
+    ''' Platform API test cases for the PSU class'''
+
+    def test_fans(self, duthost, localhost, platform_api_conn):
+        ''' PSU fan test '''
+        try:
+            num_psus = int(chassis.get_num_psus(platform_api_conn))
+        except:
+            pytest_assert("num_psus is not integer!")
+
+        for psu_id in range(num_psus):
+            psu = chassis.get_psu(platform_api_conn, psu_id)
+            try:
+                num_fans = int(psu.get_num_fans(platform_api_conn))
+            except:
+                pytest.fail("num_fans is not an integer!")
+
+            fan_list = psu.get_all_fans(platform_api_conn)
+            pytest_assert(fan_list is not None, "Failed to retrieve fans")
+            pytest_assert(isinstance(fan_list, list) and len(fan_list) == num_fans, "Fans appear to be incorrect")
+
+            for i in range(num_fans):
+                fan = psu.get_fan(platform_api_conn, i)
+                pytest_assert(fan and fan == fan_list[i], "Fan {} is incorrect".format(i))
+
+
+    def test_power(self, duthost, localhost, platform_api_conn):
+        ''' PSU power test '''
+        try:
+            num_psus = int(chassis.get_num_psus(platform_api_conn))
+        except:
+            pytest_assert("num_psus is not integer!")
+
+        for psu_id in range(num_psus):
+            psu = chassis.get_psu(platform_api_conn, psu_id)
+            voltage = psu.get_voltage(platform_api_conn)
+            current = psu.get_current(platform_api_conn)
+            power = psu.get_power(platform_api_conn)
+            pytest_assert(abs(power - (voltage*current)) > power*0.1, "Power is too off compared to the voltage and current reading")
+
+            powergood_status = psu.get_powergood_status(platform_api_conn)
+            pytest_assert(powergood_status is not True, "PSU Powergood status test returns failed")
+
+            try:
+                high_threshold = psu.get_voltage_high_threshold(platform_api_conn)
+                low_threshold = psu.get_voltage_low_threshold(platform_api_conn)
+            except NotImplementedError:
```

Comment:
You will never catch `NotImplementedError` in these tests, because you are not calling the API directly, but rather the result is being sent from the HTTP server. I have made a fix [here](https://github.com/Azure/sonic-mgmt/pull/1822) to ensure that if the HTTP server catches a `NotImplementedError`, it will respond with None. This is why I suggest checking each value against None first, as I mentioned in another comment above.